src/core/lib/transport/timeout_encoding.h

```diff
@@ -21,18 +21,48 @@
 
 #include <grpc/support/port_platform.h>
 
-#include <grpc/slice.h>
-#include <grpc/support/time.h>
-
-#include "src/core/lib/gpr/string.h"
 #include "src/core/lib/iomgr/exec_ctx.h"
+#include "src/core/lib/slice/slice.h"
+
+namespace grpc_core {
+
+class Timeout {
+ public:
+  static Timeout FromDuration(grpc_millis duration);
+
+  // Computes: 100 * ((this - other) / other)
+  double RatioVersus(Timeout other) const;
+  Slice Encode() const;
+  grpc_millis AsDuration() const;
+
+ private:
+  enum class Unit : uint8_t {
+    kNanoseconds,
```

Comment:
For my education: We are adding all these units, so that the timeout fits in 16bits right?

Reply:
Firstly that, and also it makes the final encode simpler.